### DISCUSSION TITLE
Can it calculate hyperfine structure?

### DISCUSSION THREAD

**Discussion by balabalabalalaba** [nikolasibalic/ARC-Alkali-Rydberg-Calculator#104] [Q&A] [answered]
I like this program, can it calculate hyperfine structure?

**Comment by balabalabalalaba** [comment]
Such as the spontaneous decay from one HF state to another and the transition freqs of HF states

**Reply by nikolasibalic** [reply]
Yes, for alkali atoms, as long as there is data available (and program comes with a large compilation of literature values).
You can find examples in [this notebook](https://arc-alkali-rydberg-calculator.readthedocs.io/en/latest/ARC_3_1_additions.html).
And using provided functions you can easily construct more complicated calculations.

**Comment by nikolasibalic** [comment, marked as answer]
Yes, for alkali atoms, as long as there is data available (and program comes with a large compilation of literature values).
You can find examples in [this notebook](https://arc-alkali-rydberg-calculator.readthedocs.io/en/latest/ARC_3_1_additions.html).
And using provided functions you can easily construct more complicated calculations.

**Reply by balabalabalalaba** [reply]
Thanks for your reply. As as a user, I think it will be better if it can directly give the dipole of 2 states whose $m_j$ or $m_f$ are unknown(such as the "dipole of 2P3/2->1S1/2").

**Reply by nikolasibalic** [reply]
Hi @balabalabalalaba , thank you for feedback. That is also supported. I recommend you to check [the section of documentation on calculating transitions between the states](https://arc-alkali-rydberg-calculator.readthedocs.io/en/latest/alkali_atom_functions.html#transitions-between-states). In particular there you have reduced dipole matrix elements such as `getReducedMatrixElementJ` method, which might be precisely what you are searching for.
